source/MaterialXCore/Version.cpp

```diff
@@ -1148,9 +1148,10 @@ void Document::upgradeVersion()
                     string channelString = channelsInput ? channelsInput->getValueString() : EMPTY_STRING;
                     size_t sourceChannelCount = CHANNEL_COUNT_MAP.at(inInput->getType());
                     size_t destChannelCount = CHANNEL_COUNT_MAP.at(node->getType());
-                    if (inInput->hasValue())
+                    const bool isDisconnected = inInput->getNodeName().empty() && inInput->getNodeGraphString().empty() && inInput->getInterfaceName().empty();
```

Comment:
Question to @jstone-lucasfilm : Should the `value` be removed as it will be caught in validation ? I've left it there so it preserves existing behaviour.

Reply:
Since the value has no effect on the rendered result, and the connection will take precedence, I would recommend that we simply remove the value in the upgrade to 1.39.

Reply:
Either approach seems fine here, as long as the connection continues to take precedence in the upgraded 1.39 document.

Reply:
I've removed it rather than preserve an invalid configuration.